### PR TITLE
Move hetero plugin

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Renamed hetero_plugin and moved it to new place

### Tickets:
 - 71047

### CODE REVIEW

File: docs/IE_DG/Deep_Learning_Inference_Engine_DevGuide.md [hunk 1]
```diff
@@ -70,7 +70,7 @@ The table below shows the plugin libraries and additional dependencies for Linux
...
 | GNA    | `libGNAPlugin.so`           | `libgna.so`,                                                | `GNAPlugin.dll`          | `gna.dll`                                                                                              |  Is not supported            |  -                                          |
-| HETERO | `libHeteroPlugin.so`        | Same as for selected plugins                                | `HeteroPlugin.dll`       | Same as for selected plugins                                                                           | `libHeteroPlugin.so`      |  Same as for selected plugins               |
+| HETERO | `libhetero_plugin.so`        | Same as for selected plugins                                | `hetero_plugin.dll`       | Same as for selected plugins                                                                           | `libhetero_plugin.so`      |  Same as for selected plugins               |
```
Comment (ilya-lavrenov) [inline docs/IE_DG/Deep_Learning_Inference_Engine_DevGuide.md:73]: why not `ov_hetero_plugin.so`?

File: tools/deployment_manager/configs/darwin.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
         "runtime/lib/intel64/libinference_engine_transformations.dylib",
```
Comment (ilya-lavrenov) [inline tools/deployment_manager/configs/darwin.json:20]: can we also remove prefixes in file names like `hetero_plugin.hpp`? I suppose all plugin should follow this scheme

File: docs/IE_DG/Deep_Learning_Inference_Engine_DevGuide.md (hunk 1 above)
Comment (ilyachur) [reply docs/IE_DG/Deep_Learning_Inference_Engine_DevGuide.md:73]: Right! Will update

File: tools/deployment_manager/configs/darwin.json (hunk 2 above)
Comment (ilyachur) [reply tools/deployment_manager/configs/darwin.json:20]: Do you mean inference_engine prefix?

File: src/plugins/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+if(ENABLE_HETERO)
+    add_subdirectory(hetero_plugin)
```
Comment (ilya-lavrenov) [inline src/plugins/CMakeLists.txt:6]: why not `hetero`?

Comment (ilyachur) [reply src/plugins/CMakeLists.txt:6]: Right. Fixed

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**